Rubberduck.Core/UI/CodeExplorer/Commands/UpdateFromFileCommand.cs

```diff
@@ -53,6 +68,39 @@ protected override void ImportFiles(ICollection<string> filesToImport, IVBProjec
                 return;
             }
 
+            var documentFiles = moduleNames
+                .Select(kvp => kvp.Key)
+                .Where(filename => Path.GetExtension(filename) != null
+                              && ComponentTypeForExtension.TryGetValue(Path.GetExtension(filename), out var componentType)
+                              && componentType == ComponentType.Document)
+                .ToHashSet();
+
+            //We can only insert inte existing documents.
+            if (!documentFiles.All(filename => modules.ContainsKey(filename)))
+            {
+                //TODO: report this to the user.
+                return;
+            }
+
+            //We must not remove document modules.
+            foreach (var filename in documentFiles)
+            {
+                modules.Remove(filename);
```

Comment:
This contradicts the comment saying we must not remove it?

Reply:
No, here we remove the docoment modules from the collection of modules that will be removed from the project.